src/libraries/Microsoft.PowerFx.Core/Syntax/NamedFormula.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Core.Syntax.Nodes;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Syntax
+{
+    /// <summary>
+    /// This encapsulates a named formula: its original script, the parsed result, and any parse errors.
+    /// </summary>
+    internal sealed class NamedFormula
+    {
+        public readonly string Script;
+
+        // The language settings used for parsing this script.
+        // May be null if the script is to be parsed in the current locale.
+        public readonly ILanguageSettings Loc;
+
+        private List<TexlError> _errors;
+
+        internal Dictionary<DName, TexlNode> _formulasResult;
```

Comment:
consider something more complex, passing along the substring corresponding to the rule, such as Dict<DName, (TexlNode node, string script)>

Reply:
Or provide like a `TryGetScript(DName, out string)` method maybe

Reply:
Added a method to get the substring.